File: kline-frontend/src/types/kline.ts
export interface KLine {
  date: string;
  open: number;
  close: number;
  high: number;
  low: number;
  volume: number;
  amount: number;
}

export interface StockInfo {
  code: string;
  name: string;
}

export interface IndicatorData {
  ma?: { [key: string]: (number | null)[] };   // ma5, ma10, ma20, ma60
  macd?: { dif: (number | null)[]; dea: (number | null)[]; macd: (number | null)[] };
  kdj?: { k: (number | null)[]; d: (number | null)[]; j: (number | null)[] };
  rsi?: { [key: string]: (number | null)[] };  // rsi6, rsi12, rsi24
  boll?: { mid: (number | null)[]; upper: (number | null)[]; lower: (number | null)[] };
}

export interface KLineResponse {
  code: number;
  message: string;
  data: {
    stock_info: StockInfo;
    klines: KLine[];
    count: number;
    period?: string;
    indicators?: IndicatorData;
  };
}

File: kline-backend/app/schemas/kline.py
"""K线相关Schema"""
from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from enum import Enum


class KLinePeriod(str, Enum):
    """K线周期枚举"""
    DAY = "day"
    WEEK = "week"
    MONTH = "month"
    YEAR = "year"


class KLineData(BaseModel):
    """单根K线数据"""
    date: str
    open: float
    close: float
    high: float
    low: float
    volume: float
    amount: float = 0.0


class StockBasicInfo(BaseModel):
    """股票基础信息"""
    code: str
    name: str


class KLineResponse(BaseModel):
    """K线数据响应"""
    stock_info: StockBasicInfo
    klines: List[KLineData]
    count: int
    period: str = "day"
    indicators: Optional[Dict[str, Any]] = None
